### ISSUE TITLE
[Bug] Two fatal errors in src/models/multiview_matchers/par_comatcher.py

### ISSUE THREAD

**Issue by jinqin-99** [EATMustard/CoMatcher#5]
<html>
<body>

Hello, you have done an excellent job！But I found several bugs in the code when reproducing it! I hope you can improve the code.

<p><strong>Environment</strong></p>
<ul>
<li>File: <code>src/models/multiview_matchers/par_comatcher.py</code></li>
<li>Framework: PyTorch</li>
</ul>
<hr>
<h3>Bug 1 — <code>AttributeError</code>: <code>self.confidence_estimator</code> is never defined on the main model</h3>
<p><strong>Severity:</strong> 🔴 Fatal — crashes on every inference call</p>
<p><strong>Location:</strong> <code>_forward()</code>, inference branch</p>
<pre><code class="language-python"># Line in _forward:
if not self.training:
    confidence = torch.sigmoid(self.confidence_estimator(s_desc))  # ← AttributeError
    AP_mask = get_AP_mask(confidence, data["S2S_matches"])
</code></pre>

<hr>
<h3>Bug 2 — <code>RuntimeError</code>: boolean indexing in <code>get_AP_mask</code> collapses the batch dimension</h3>
<p><strong>Severity:</strong> 🔴 Fatal — crashes on every call to <code>get_AP_mask</code></p>
<p><strong>Location:</strong> <code>get_AP_mask()</code>, inner loop</p>
<pre><code class="language-python">for i in range(N):
    for j in range(N):
        matches_ij = s2s_matches[:, i, j]           # shape: (b, n)
        valid = (matches_ij != -1) &amp; (conf_mask[:, j] &gt; 0)  # shape: (b, n)

        src_indices = matches_ij[valid] + j * n      # ← boolean index flattens batch dim!
        # src_indices is now 1-D, shape mismatch with (b, n) in torch.where below
        tgt_indices = torch.where(valid, ..., AP_mask[:, i])  # shape: (b, n) → CRASH
</code></pre>

</body>
</html>
